ICE MRE: Failed to normalize Alias(Projection) for DeepModel during derive(PartialEq) on struct containing custom enum

Sorry to bother you again and appreciate your time
Here's the Minimal Reproducible Example:

### Summary
Creusot panics with an ICE when a struct/enum contains a field of a custom enum type that does not implement `DeepModel`. The panic is triggered during the standard Rust `derive(PartialEq)` macro expansion for the outer type.

### Minimal Reproduction
cargo.toml
[package]
name = "creusot-ice-test"
version = "0.1.0"
edition = "2021"

[dependencies]
creusot-std = "0.10.0"

src/lib.rs
```rust
use creusot_std::prelude::*;
use std::clone::Clone;
use std::cmp::PartialEq;
use std::cmp::Eq;

#[derive(Clone, Copy, PartialEq, Eq)]
pub enum Inner { A, B }

#[derive(Clone, Copy, PartialEq, Eq)]
pub enum Outer { Bar { inner: Inner } }

#[cfg(creusot)]
impl Outer {
    #[ensures(match (*self) {
        Outer::Bar { inner, .. } => result == (match inner {
            Inner::A => true,
            Inner::B => false,
        }),
    })]
    pub fn test_nested_match(&self) -> bool {
        match self {
            Outer::Bar { inner } => match *inner {
                Inner::A => true,
                Inner::B => false,
            },
        }
    }
}
```

### Steps to Reproduce
1. Create a new crate with `creusot-contracts` as a dependency.
2. Paste the code above into `src/lib.rs`.
3. Run `cargo creusot prove`.

### Exact Error Output

```text
error[E0277]: the trait bound `Inner: creusot_std::model::DeepModel` is not satisfied
 --> src/lib.rs:8:24
  |
7 | #[derive(Clone, Copy, PartialEq, Eq)]
  |                       --------- in this derive macro expansion
8 | pub enum Outer { Bar { inner: Inner } }
  |                        ^^^^^^^^^^^^ unsatisfied trait bound
  |
help: the trait `creusot_std::model::DeepModel` is not implemented for `Inner`
 --> src/lib.rs:6:1
  |
6 | pub enum Inner { A, B }
  | ^^^^^^^^^^^^^^
  = note: required for `&Inner` to implement `creusot_std::model::DeepModel`

error: internal compiler error: /rustc-dev/de6d33c033441c5880b863f94d7a3ec8cad141bd/compiler/rustc_middle/src/ty/normalize_erasing_regions.rs:171:13: Failed to normalize Alias(Projection, AliasTy { args: [&'{erased} Inner], def_id: DefId(20:978 ~ creusot_std[0ec0]::model::Dee

<img width="1247" height="989" alt="Image" src="https://github.com/user-attachments/assets/3fd4c615-1461-4f35-94ff-2009d6cc2950" />

pModel::DeepModelTy), .. }) in typing_env=TypingEnv { typing_mode: Analysis { defining_opaque_types_and_generators: [] }, param_env: ParamEnv { caller_bounds: [] } }, maybe try to call `try_normalize_erasing_regions` instead

thread 'rustc' panicked at /rustc-dev/de6d33c033441c5880b863f94d7a3ec8cad141bd/compiler/rustc_middle/src/ty/normalize_erasing_regions.rs:171:13:
Box<dyn Any>
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

--- a/creusot/src/translation/function/terminator.rs
+++ b/creusot/src/translation/function/terminator.rs
@@ -88,7 +88,7 @@ impl<'tcx> BodyTranslator<'_, 'tcx> {
                             span,
                         );
                         if let Err(errs) = res {
-                            infcx.err_ctxt().report_fulfillment_errors(errs);
+                            infcx.err_ctxt().report_fulfillment_errors(errs).raise_fatal();
                         }
 
                         let tr_res = TraitResolved::resolve_item(

--- a/tests/should_fail/bug/1610-crash.rs
+++ b/tests/should_fail/bug/1610-crash.rs
@@ -1,9 +1,8 @@
 // Test a crash when trying to implement collect
 // This test may become should_succeed in the future.
 extern crate creusot_std;
-use creusot_std::prelude::*;
 
-struct P();
+pub struct P();
 
 impl ::std::iter::Iterator for P {
     type Item = ();

--- a/tests/should_fail/bug/1610-crash.stderr
+++ b/tests/should_fail/bug/1610-crash.stderr
@@ -1,28 +1,9 @@
-warning: unused import: `creusot_std::prelude::*`
- --> 1610-crash.rs:4:5
-  |
-4 | use creusot_std::prelude::*;
-  |     ^^^^^^^^^^^^^^^^^^^^^^^
-  |
-  = note: `#[warn(unused_imports)]` (part of `#[warn(unused)]`) on by default
-
-warning: struct `P` is never constructed
- --> 1610-crash.rs:6:8
-  |
-6 | struct P();
-  |        ^
-  |
-  = note: `#[warn(dead_code)]` (part of `#[warn(unused)]`) on by default
-
 error[E0277]: the trait bound `B: creusot_std::prelude::FromIteratorSpec<()>` is not satisfied
-  --> 1610-crash.rs:15:9
+  --> 1610-crash.rs:14:9
    |
-15 |         std::iter::empty().collect()
+14 |         std::iter::empty().collect()
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `creusot_std::prelude::FromIteratorSpec<()>` is not implemented for `B`
 
-error: could not resolve trait instance for creusot_std::prelude::FromIteratorSpec::from_iter_post[B, <std::iter::Empty<()> as std::iter::Iterator>::Item]
- --> creusot-std/src/std/iter.rs:173:88
-
-error: aborting due to 2 previous errors; 2 warnings emitted
+error: aborting due to 1 previous error
 
 For more information about this error, try `rustc --explain E0277`.

--- a/tests/should_fail/bug/2040.rs
+++ b/tests/should_fail/bug/2040.rs
@@ -1,0 +1,5 @@
+extern crate creusot_std;
+
+pub fn f<T: Eq>(x: T) -> bool {
+    x == x
+}

--- a/tests/should_fail/bug/2040.stderr
+++ b/tests/should_fail/bug/2040.stderr
@@ -1,0 +1,9 @@
+error[E0277]: the trait bound `T: creusot_std::model::DeepModel` is not satisfied
+ --> 2040.rs:4:5
+  |
+4 |     x == x
+  |     ^^^^^^ the trait `creusot_std::model::DeepModel` is not implemented for `T`
+
+error: aborting due to 1 previous error
+
+For more information about this error, try `rustc --explain E0277`.

